types/webpack-config-utils/index.d.ts

```diff
@@ -0,0 +1,64 @@
+// Type definitions for webpack-config-utils 2.3
+// Project: https://github.com/kentcdodds/webpack-config-utils#readme
+// Definitions by: Martin Hochel <https://github.com/hotell>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.8
+
+/*~ You can declare properties of the module using const, let, or var */
+// declare const getIfUtils: WebpackConfigUtils.GetIfUtils;
+// declare const removeEmpty: WebpackConfigUtils.RemoveEmpty;
+// declare const propIf: WebpackConfigUtils.PropIf;
+// declare const propIfNot: WebpackConfigUtils.PropIfNot;
+
+declare const api: WebpackConfigUtils.API;
+export = api;
+
+declare namespace WebpackConfigUtils {
+    type Falsy = false | '' | undefined | null | 0;
+    type DefinedObjKeys<T> = ({ [P in keyof T]: T[P] extends undefined ? never : P })[keyof T];
+    type NonEmptyObject<T, P extends DefinedObjKeys<T> = DefinedObjKeys<T>> = { [PP in P]: T[PP] };
+
+    type EnvVars = 'production' | 'prod' | 'test' | 'development' | 'dev';
+    interface RemoveEmpty {
+        <T>(input: Array<T | undefined>): T[];
+        <T>(input: { [P in keyof T]: T[P] }): NonEmptyObject<T>;
+    }
+    type GetIfUtils = <E extends EnvVars | string>(
+        env: { [P in E]: boolean | string } | E,
+        vars?: Array<EnvVars | string>
+    ) => IfUtils;
+    // @TODO
```

Comment:
Hey folks! Any ideas how to use this?

Generics are not properly inferred when I wanna use this conditional mapping

```ts
type Falsy = false | '' | undefined | null | 0;
type PropIf = <A, I, E>(add: A, value: I, alternate: E) => A extends Falsy ? E : I;

//...

 // $ExpectType "string"
propIf(true, 'value', 'alternate'); // 'value'

// if generics are explicitly provided 👉

 // $ExpectType "value"
propIf<true, 'value', 'alternate'>(true, 'value', 'alternate');
```